src/transaction/transaction.ts

```diff
@@ -421,22 +421,32 @@ export async function populateAppCallResources(atc: algosdk.AtomicTransactionCom
     }
 
     if (type === 'account') {
-      txns[txnIndex].txn.appAccounts?.push(algosdk.decodeAddress(reference as string))
+      txns[txnIndex].txn.appAccounts = [...(txns[txnIndex].txn.appAccounts ?? []), algosdk.decodeAddress(reference as string)]
     } else if (type === 'app') {
-      txns[txnIndex].txn.appForeignApps?.push(Number(reference))
+      txns[txnIndex].txn.appForeignApps = [...(txns[txnIndex].txn.appForeignApps ?? []), Number(reference)]
     } else if (type === 'box') {
       const { app, name } = reference as algosdk.modelsv2.BoxReference
-      txns[txnIndex].txn.boxes?.push({ appIndex: Number(app), name: name })
+      txns[txnIndex].txn.boxes = [...(txns[txnIndex].txn.boxes ?? []), { appIndex: Number(app), name }]
+
+      if (app !== 0) {
+        // Add the app if it is not already available
+        let appAlreadyAvailable = false
+        txns.forEach((t) => {
+          if (t.txn.appIndex === Number(reference)) appAlreadyAvailable = true
+          if (t.txn.appForeignApps?.includes(Number(reference))) appAlreadyAvailable = true
```

Comment:
probably clearer to do a single if with an `||` instead of two if statements & two assignments

Reply:
Yeah true. I simplified the assignment and switched to `for...of` so we can break the loop in 75498cb696cc72ea9ded3e992138884801645c46

Reply:
github needs to add the fire emoji